lib/client.js

```diff
@@ -143,11 +143,19 @@ class Client extends EventEmitter {
       }
     }
 
-    this[kQueue] = Q((request, cb) => {
+    const dispatch = (request, cb) => {
       if (this.destroyed) {
         return cb(new Error('The client is destroyed'))
       }
 
+      if (this.socket.destroyed) {
+        // TODO: This doesn't work
+        // this[kQueue].unshift(request, cb)
+        // this[kQueue].pause()
+        // cb()
```

Comment:
@mcolline: I need help here... fastq is doing some magic here where it's not entirely clear how to reschedule a task.